src/layers/attention.rs

```diff
@@ -52,16 +52,143 @@ pub enum QkvMode {
     MergedSplitAfter(QkvSplit),
 }
 
+/// ALiBi: Linear biases for attention [Press et al., 2022](https://arxiv.org/abs/2108.12409).
+pub struct AttentionLinearBiases {
+    slopes: Tensor,
+    is_causal: bool,
+    is_inverted: bool,
+}
+
+impl AttentionLinearBiases {
+    pub fn new(n_attention_heads: usize, is_causal: bool, is_inverted: bool) -> Result<Self> {
+        let slopes = Self::calculate_slopes(n_attention_heads)?;
+        Ok(Self {
+            slopes,
+            is_causal,
+            is_inverted,
+        })
+    }
+
+    /// Calculate the linear bias slopes for a given number
+    /// of attention heads.
+    ///
+    /// * n_attention_heads - Number of attention heads.
+    ///
+    /// Returns: Head slope tensor.
+    /// *Shape:* `(1, heads, 1, 1)`
+    #[allow(clippy::unnecessary_cast)]
+    fn calculate_slopes(n_attention_heads: usize) -> Result<Tensor> {
+        fn slopes_with_step(n_attention_heads: usize, step: usize) -> Result<Tensor> {
+            let ratio = 2.0f32.powf(-8.0f32 / n_attention_heads as f32);
+            let slope = (1..(1 + n_attention_heads))
```

Comment:
```suggestion
            let slope = (1..n_attention_heads + 1)
```

looks more canonical.